api/v1alpha1/gateway_types.go

```diff
@@ -325,6 +325,15 @@ type RouteBindingSelector struct {
 	//
 	// +optional
 	RouteSelector *metav1.LabelSelector `json:"routeSelector,omitempty" protobuf:"bytes,2,opt,name=routeSelector"`
+	// Kind specifies the kind of resource to select with the above selectors.
+	//
+	// Kind MUST correspond to route resources that are compatible with the
+	// application protocol specified in the Listener's Protocol field.
+	//
+	// Support: Core
+	//
+	// +required
+	Kind metav1.GroupVersionKind `json:"kind" protobuf:"bytes,3,opt,name=kind"`
```

Comment:
Does k8s has other apis using GVK in them? It feels weird to have 
```
kind:
   kind: 
   ...
```

but not sure what others are doing

Reply:
Good question - there's not much use that've seen. The only other example I've found is in `AdmissionReview.Request`, which also use `Kind`. It does look like `kind` is the go-to variable name for GVK as well, so this shouldn't be unfamiliar. The simple alternative would be to call it `GroupVersionKind` which seems like it's already the mental translation of "kind" in k8s context.

Reply:
You can embed and pull up using the +inline directive.

Reply:
Do we want to do that? It seems like having a contained GVK is a good thing and follows the admission.k8s.io precedent

Reply:
hmm, shame on me to not expand the diff and comment out of context:

it's better to have 

```
  routes:
    selector: ...
    kind: blah
    group:
```

which would be 

```
type GatewayRoutes struct {
  Selector v1.LabelSelector
  Kind GroupVersionKind `json:"inline"` // I think this is the syntax?
}
```

Reply:
I think I generally agree - however with e.g. AdmissionReview, Kind is left as a distinct resource:
```
kind:
  kind: ...
  group: ...
  version: ...
```
Which I agree is not as clean, but it's currently the only precedent. I'm happy to go either way

Reply:
One thing to note - inlined struct fields don't have independent descriptions in the generated CRD yaml. Explanations of these fields would need to move to RouteBindingSelector (which is probably ok)